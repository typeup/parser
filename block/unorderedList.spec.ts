import { Error, IO } from "@cogneco/mend"
import { Source } from "../Source"
import { parse } from "."

describe("block.unorderedList", () => {
	it("single level", () => {
		const result = parse(new Source(IO.StringReader.create("- item A\n- item B\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
	it("paragraph after", () => {
		const result = parse(new Source(IO.StringReader.create("- item A\n- item B\nThis is a simple list.\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
	it("multi level", () => {
		const result = parse(new Source(IO.StringReader.create("- item A\n- item B\n\t- item B1\n\t- item B2\n- item C\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
// start here
	it("multi level", () => {
		const result = parse(new Source(IO.StringReader.create("1. itemized A\n1. itemized B\n\t1. itemized B1\n\t1. itemized B2\n1. itemized C\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})

	it("multi level", () => {
		const result = parse(new Source(IO.StringReader.create("- item A\n- item B\n\t1. itemized B1\n\t1. itemized B2\n- item C\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
	it("multi level", () => {
		const result = parse(new Source(IO.StringReader.create("1. itemized A\n1. itemized B\n\t- item B1\n\t- item B2\n1. itemized C\n"), new Error.ConsoleHandler())) || []
		expect(result.map(node => node.toObject())).toMatchSnapshot()
	})
})
