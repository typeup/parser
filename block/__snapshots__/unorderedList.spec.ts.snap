// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block.unorderedList multi level 1`] = `
Array [
  Object {
    "class": "Block.UnorderedList",
    "content": Array [
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "item A
",
              },
            ],
          },
        ],
      },
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "item B
",
              },
            ],
          },
          Object {
            "class": "Block.UnorderedList",
            "content": Array [
              Object {
                "class": "Block.ListItem",
                "content": Array [
                  Object {
                    "class": "Block.Paragraph",
                    "content": Array [
                      Object {
                        "class": "Inline.Text",
                        "value": "item B1
",
                      },
                    ],
                  },
                ],
              },
              Object {
                "class": "Block.ListItem",
                "content": Array [
                  Object {
                    "class": "Block.Paragraph",
                    "content": Array [
                      Object {
                        "class": "Inline.Text",
                        "value": "item B2
",
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
      },
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "item C
",
              },
            ],
          },
        ],
      },
    ],
  },
]
`;

exports[`block.unorderedList multi level 2`] = `
Array [
  Object {
    "class": "Block.OrderedList",
    "content": Array [
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "itemized A
",
              },
            ],
          },
        ],
      },
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "itemized B
",
              },
            ],
          },
          Object {
            "class": "Block.OrderedList",
            "content": Array [
              Object {
                "class": "Block.ListItem",
                "content": Array [
                  Object {
                    "class": "Block.Paragraph",
                    "content": Array [
                      Object {
                        "class": "Inline.Text",
                        "value": "itemized B1
",
                      },
                    ],
                  },
                ],
              },
              Object {
                "class": "Block.ListItem",
                "content": Array [
                  Object {
                    "class": "Block.Paragraph",
                    "content": Array [
                      Object {
                        "class": "Inline.Text",
                        "value": "itemized B2
",
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
      },
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "itemized C
",
              },
            ],
          },
        ],
      },
    ],
  },
]
`;

exports[`block.unorderedList multi level 3`] = `
Array [
  Object {
    "class": "Block.UnorderedList",
    "content": Array [
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "item A
",
              },
            ],
          },
        ],
      },
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "item B
",
              },
            ],
          },
          Object {
            "class": "Block.OrderedList",
            "content": Array [
              Object {
                "class": "Block.ListItem",
                "content": Array [
                  Object {
                    "class": "Block.Paragraph",
                    "content": Array [
                      Object {
                        "class": "Inline.Text",
                        "value": "itemized B1
",
                      },
                    ],
                  },
                ],
              },
              Object {
                "class": "Block.ListItem",
                "content": Array [
                  Object {
                    "class": "Block.Paragraph",
                    "content": Array [
                      Object {
                        "class": "Inline.Text",
                        "value": "itemized B2
",
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
      },
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "item C
",
              },
            ],
          },
        ],
      },
    ],
  },
]
`;

exports[`block.unorderedList multi level 4`] = `
Array [
  Object {
    "class": "Block.OrderedList",
    "content": Array [
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "itemized A
",
              },
            ],
          },
        ],
      },
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "itemized B
",
              },
            ],
          },
          Object {
            "class": "Block.UnorderedList",
            "content": Array [
              Object {
                "class": "Block.ListItem",
                "content": Array [
                  Object {
                    "class": "Block.Paragraph",
                    "content": Array [
                      Object {
                        "class": "Inline.Text",
                        "value": "item B1
",
                      },
                    ],
                  },
                ],
              },
              Object {
                "class": "Block.ListItem",
                "content": Array [
                  Object {
                    "class": "Block.Paragraph",
                    "content": Array [
                      Object {
                        "class": "Inline.Text",
                        "value": "item B2
",
                      },
                    ],
                  },
                ],
              },
            ],
          },
        ],
      },
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "itemized C
",
              },
            ],
          },
        ],
      },
    ],
  },
]
`;

exports[`block.unorderedList paragraph after 1`] = `
Array [
  Object {
    "class": "Block.UnorderedList",
    "content": Array [
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "item A
",
              },
            ],
          },
        ],
      },
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "item B
",
              },
            ],
          },
        ],
      },
    ],
  },
  Object {
    "class": "Block.Paragraph",
    "content": Array [
      Object {
        "class": "Inline.Text",
        "value": "This is a simple list.
",
      },
    ],
  },
]
`;

exports[`block.unorderedList single level 1`] = `
Array [
  Object {
    "class": "Block.UnorderedList",
    "content": Array [
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "item A
",
              },
            ],
          },
        ],
      },
      Object {
        "class": "Block.ListItem",
        "content": Array [
          Object {
            "class": "Block.Paragraph",
            "content": Array [
              Object {
                "class": "Inline.Text",
                "value": "item B
",
              },
            ],
          },
        ],
      },
    ],
  },
]
`;
